filters/include/pcl/filters/impl/extract_indices.hpp

```diff
@@ -55,9 +55,9 @@ pcl::ExtractIndices<PointT>::filterDirectly (PointCloudPtr &cloud)
 
   std::vector<pcl::PCLPointField> fields;
   pcl::for_each_type<FieldList> (pcl::detail::FieldAdder<PointT> (fields));
-  for (int rii = 0; rii < static_cast<int> (removed_indices_->size ()); ++rii)  // rii = removed indices iterator
+  for (const auto& rii : (*removed_indices_)) // rii = removed indices iterator
   {
-    std::size_t pt_index = (std::size_t) (*removed_indices_)[rii];
+    uindex_t pt_index = (uindex_t) rii;
```

Comment:
`const auto` would be better, even preventing comparison between unsigned and signed in line 61, right?

Reply:
Do you mean like this `const auto pt_index = rii;`? Then `pt_index` would be exactly the same as `rii` and potentially signed.

Reply:
Ah, I forgot that the PR to make `pointcloud.size()` return `index_t` is not mature yet.